byway/src/main/webapp/destinations.js

```diff
@@ -132,15 +131,37 @@ function addMarker(searchBox,map){
 /** 
 * fetches start location and destinations from DestinationsServlet and adds to DOM
 */
-function updateLocations(locationData){
-    document.getElementById('start-location').innerText = "Start Location :" + locationData.start;
+function updateLocations(locationData, map){
     const container = document.getElementById('destinations-container');
-    container.innerText = "Destinations:";
+    container.innerText="";
     let destinationArray= locationData.destinations;
     destinationArray.forEach((destination) => {
-      let destinationToAdd = document.createElement('p');
-      destinationToAdd.innerText = destination;
-      container.appendChild(destinationToAdd);
+      const request = {
+        query: destination,
+        fields: ["name", "photos", "formatted_address", "rating", "business_status"]
+      };
+      let service = new google.maps.places.PlacesService(map);
+      service.findPlaceFromQuery(request, (results, status) => {
+        if (status === google.maps.places.PlacesServiceStatus.OK) {
+          let destinationToAdd = document.createElement('p');
```

Comment:
Consider making a new function to handle this task of adding this destination along with the elements you append. Also, what if the status isn't `OK`?

Reply:
done